code/src/main/kotlin/com/expediagroup/sdk/v2/core/apache/util/ApacheHttpTransportUtil.kt

```diff
@@ -1,35 +1,66 @@
 package com.expediagroup.sdk.v2.core.apache.util
 
-import com.expediagroup.sdk.v2.core.trait.common.IdTrait
 import com.expediagroup.sdk.v2.core.trait.configuration.ClientConfiguration
 import com.google.api.client.http.apache.v2.ApacheHttpTransport
-import java.util.*
-import java.util.concurrent.ConcurrentHashMap
-import java.util.concurrent.ConcurrentMap
 
+/**
+ * Creates an `ApacheHttpTransport` instance based on the provided `ClientConfiguration`.
+ *
+ * @param configuration The `ClientConfiguration` used to configure the `ApacheHttpTransport` instance.
+ * @return A new instance of `ApacheHttpTransport` configured according to the provided `ClientConfiguration`.
+ */
```

Comment:
I feel that the implementation here is overcomplicated with the lambda style, getting a singleton object should be more straightforward. You can simply declare a private object with `getInstance()` method.

```kotlin
private object ApacheHttpTransportSingleton {
    private var transport: ApacheHttpTransport? = null

    fun getInstance(configuration: ClientConfiguration): ApacheHttpTransport {
        if (transport == null) {
            transport = ApacheHttpTransport(createHttpClient(configuration))
        }

        return transport!!
    }
}
```

Reply:
Done.